extension/src/service/klarna-service.js

```diff
@@ -0,0 +1,86 @@
+const {
+  ADYEN_PERCENTAGE_MINOR_UNIT,
+  DEFAULT_PAYMENT_LANGUAGE,
+  KLARNA_DEFAULT_LINE_ITEM_NAME
+} = require('../config/klarna-constants')
+
+function createLineItems (payment, cart) {
+  const lineItems = []
+  const locales = _getLocales(cart, payment)
+
+  cart.lineItems.forEach((item) => {
+    lineItems.push(_createAdyenLineItemFromLineItem(item, locales))
+  })
+
+  cart.customLineItems.forEach((item) => {
+    lineItems.push(_createAdyenLineItemFromCustomLineItem(item, locales))
+  })
+
+  lineItems.push(_createShippingInfoAdyenLineItem(cart.shippingInfo))
+
+  return lineItems
+}
+
+function _getLocales (cart, payment) {
+  const locales = []
+  let paymentLanguage = payment.custom && payment.custom.fields['languageCode']
+  if (!paymentLanguage)
+    paymentLanguage = cart.locale
+  if (paymentLanguage)
+    locales.push(paymentLanguage)
+  if (!paymentLanguage || paymentLanguage !== DEFAULT_PAYMENT_LANGUAGE)
+    locales.push(DEFAULT_PAYMENT_LANGUAGE)
+
+  return locales
+}
+
+function _createAdyenLineItemFromLineItem (ctpLineItem, locales) {
+  return {
+    id: ctpLineItem.variant.sku,
+    quantity: ctpLineItem.quantity,
+    description: _localizeOrFallback(ctpLineItem.name, locales, KLARNA_DEFAULT_LINE_ITEM_NAME),
+    amountIncludingTax: ctpLineItem.price.value.centAmount,
+    taxPercentage: ctpLineItem.taxRate.amount * ADYEN_PERCENTAGE_MINOR_UNIT
+  }
+}
+
+function _createAdyenLineItemFromCustomLineItem (ctpLineItem, locales) {
+  return {
+    id: ctpLineItem.id,
+    quantity: ctpLineItem.quantity,
+    description: _localizeOrFallback(ctpLineItem.name, locales, KLARNA_DEFAULT_LINE_ITEM_NAME),
+    amountIncludingTax: ctpLineItem.money.centAmount,
+    taxPercentage: ctpLineItem.taxRate.amount * ADYEN_PERCENTAGE_MINOR_UNIT
+  }
+}
+
+function _createShippingInfoAdyenLineItem (shippingInfo) {
+  return {
+    id: `${shippingInfo.shippingMethodName}`,
+    quantity: 1,
```

Comment:
why it's 1?

Reply:
Because you can have only 1 shipping per cart.

Reply:
See more: https://github.com/commercetools/commercetools-payone-integration/blob/dfb14f4f17c356fa0d14b7f007aebe76065942ec/service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/model/common/AuthorizationRequestWithCart.java#L290

Reply:
This is not clear when you read the code, please add this as a comment.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/294/commits/419c4a50172ef8f2d667dc57bc5fd28eaadf0703#diff-4d2f466e11952575f6b8048a0f137b16R97